File: GNSDK/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnVideoChapter.java

package com.gracenote.gnsdk;

/** 
**/ 
 
public class GnVideoChapter extends GnDataObject {
  private long swigCPtr;

  protected GnVideoChapter(long cPtr, boolean cMemoryOwn) {
    super(gnsdk_javaJNI.GnVideoChapter_SWIGUpcast(cPtr), cMemoryOwn);
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnVideoChapter obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnVideoChapter(swigCPtr);
      }
      swigCPtr = 0;
    }
    super.delete();
  }

  public static String gnType() {
    return gnsdk_javaJNI.GnVideoChapter_gnType();
  }

  public static GnVideoChapter from(GnDataObject obj) throws com.gracenote.gnsdk.GnException {
    return new GnVideoChapter(gnsdk_javaJNI.GnVideoChapter_from(GnDataObject.getCPtr(obj), obj), true);
  }

/** 
* Video chapter's ordinal value. 
* 
*/ 
 
  public long ordinal() {
    return gnsdk_javaJNI.GnVideoChapter_ordinal(swigCPtr, this);
  }

/** 
*  Chapter's duration value in seconds such as "3600" for a 60-minute program. 
* 
*/ 
 
  public long duration() {
    return gnsdk_javaJNI.GnVideoChapter_duration(swigCPtr, this);
  }

/** 
*  Chapter's duration units value (seconds, "SEC"). 
* 
*/ 
 
  public String durationUnits() {
    return gnsdk_javaJNI.GnVideoChapter_durationUnits(swigCPtr, this);
  }

/** 
*  Official Title object. 
* 
*/ 
 
  public GnTitle officialTitle() {
    return new GnTitle(gnsdk_javaJNI.GnVideoChapter_officialTitle(swigCPtr, this), true);
  }

  public GnVideoCreditIterable videoCredits() {
    return new GnVideoCreditIterable(gnsdk_javaJNI.GnVideoChapter_videoCredits(swigCPtr, this), true);
  }

}

File: GNSDK/wrappers/gnsdk_csharp/src_wrapper/GnVideoChapter.cs

namespace GracenoteSDK {

using System;
using System.Runtime.InteropServices;

/**
* \class GnVideoChapter
*/
public class GnVideoChapter : GnDataObject {
  private HandleRef swigCPtr;

  internal GnVideoChapter(IntPtr cPtr, bool cMemoryOwn) : base(gnsdk_csharp_marshalPINVOKE.GnVideoChapter_SWIGUpcast(cPtr), cMemoryOwn) {
    swigCPtr = new HandleRef(this, cPtr);
  }

  internal static HandleRef getCPtr(GnVideoChapter obj) {
    return (obj == null) ? new HandleRef(null, IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnVideoChapter() {
    Dispose();
  }

  public override void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnVideoChapter(swigCPtr);
        }
        swigCPtr = new HandleRef(null, IntPtr.Zero);
      }
      GC.SuppressFinalize(this);
      base.Dispose();
    }
  }

  public static string GnType() {
    string ret = gnsdk_csharp_marshalPINVOKE.GnVideoChapter_GnType();
    return ret;
  }

  public static GnVideoChapter From(GnDataObject obj) {
    GnVideoChapter ret = new GnVideoChapter(gnsdk_csharp_marshalPINVOKE.GnVideoChapter_From(GnDataObject.getCPtr(obj)), true);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
    return ret;
  }

/**
* Video chapter's ordinal value.
* @ingroup GDO_ValueKeys_Misc
*/
  public uint Ordinal {
    get {
      uint ret = gnsdk_csharp_marshalPINVOKE.GnVideoChapter_Ordinal_get(swigCPtr);
      return ret;
    } 
  }

/**
*  Chapter's duration value in seconds such as "3600" for a 60-minute program.
* @ingroup GDO_ValueKeys_Video
*/
  public uint Duration {
    get {
      uint ret = gnsdk_csharp_marshalPINVOKE.GnVideoChapter_Duration_get(swigCPtr);
      return ret;
    } 
  }

/**
*  Chapter's duration units value (seconds, "SEC").
* @ingroup GDO_ValueKeys_Video
*/
  public string DurationUnits {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoChapter_DurationUnits_get(swigCPtr) );
	} 

  }

/**
*  Official Title object.
* @ingroup GDO_ChildKeys_Title
*/
  public GnTitle OfficialTitle {
    get {
      IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnVideoChapter_OfficialTitle_get(swigCPtr);
      GnTitle ret = (cPtr == IntPtr.Zero) ? null : new GnTitle(cPtr, true);
      return ret;
    } 
  }

  public GnVideoCreditEnumerable VideoCredits {
    get {
      IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnVideoChapter_VideoCredits_get(swigCPtr);
      GnVideoCreditEnumerable ret = (cPtr == IntPtr.Zero) ? null : new GnVideoCreditEnumerable(cPtr, true);
      return ret;
    } 
  }

}

}
